pkg/logurls/logurls.go

```diff
@@ -82,11 +84,11 @@ func generateSubstitutionMaps(m map[string]interface{}) []map[string]string {
 	return vmaps
 }
 
-func interpolate(val string, env map[string]string, r *regexp.Regexp) string {
+func interpolate(val string, env map[string]string, r *regexp.Regexp) models.LogTuple {
 	matches := r.FindAllStringSubmatch(val, -1)
 	if matches == nil {
 		// Finished, nothing more to substitute.
-		return val
+		return models.LogTuple{ContainerName: env[containerName], Url: val}
```

Comment:
The sink should pull the container name value from the resource JSON, not from the evaluated `CONTAINER_NAME` variable.

Reply:
How will interpolate know which container name the url is for with out looking at the evaluated `CONTAINER_NAME`?

Reply:
The log URL code should pass back all the information needed (i.e. the URL and the container name), but it should *not* depend on the `CONTAINER_NAME` variable for the value.  There is no guarantee the user will set it correctly and then bad URLs will result.

I think instead what we should do, and what is probably quickest, is to say that `CONTAINER_NAME` is a predefined variable name and set it to the CEL express to representing the list of container names. Then it can be evaluated and used as needed if a user needs it for generating the URL.

So I think what can be done is this:

- Add `CONTAINER_NAME` to the map as `CONTAINER_NAME: "cel:spec.containers.map(m, m.name)"`, overwriting whatever setting there might be specified in the `ConfigMap`.
- Document this behavior, and note that URLs will be generated for each container in the `Pod`.  This may already be noted in the documentation.

Reply:
done